aws/resource_aws_wafv2_web_acl.go

```diff
@@ -0,0 +1,843 @@
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9-_]+$`), "must contain only alphanumeric hyphen and underscore characters"),
+				),
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.ScopeCloudfront,
+					wafv2.ScopeRegional,
+				}, false),
+			},
+			"rule": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"allow": wafv2EmptySchema(),
+									"block": wafv2EmptySchema(),
+									"count": wafv2EmptySchema(),
+								},
+							},
+						},
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringLenBetween(1, 128),
+						},
+						"override_action": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"count": wafv2EmptySchema(),
+									"none":  wafv2EmptySchema(),
+								},
+							},
+						},
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"statement":         wafv2WebACLRootStatementSchema(3),
+						"visibility_config": wafv2VisibilityConfigSchema(),
+					},
+				},
+			},
+			"tags":              tagsSchema(),
+			"visibility_config": wafv2VisibilityConfigSchema(),
+		},
+	}
+}
+
+func resourceAwsWafv2WebACLCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	var resp *wafv2.CreateWebACLOutput
+
+	params := &wafv2.CreateWebACLInput{
+		Name:             aws.String(d.Get("name").(string)),
+		Scope:            aws.String(d.Get("scope").(string)),
+		DefaultAction:    expandWafv2DefaultAction(d.Get("default_action").([]interface{})),
+		Rules:            expandWafv2WebACLRules(d.Get("rule").(*schema.Set).List()),
+		VisibilityConfig: expandWafv2VisibilityConfig(d.Get("visibility_config").([]interface{})),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		params.Description = aws.String(v.(string))
+	}
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		params.Tags = keyvaluetags.New(v).IgnoreAws().Wafv2Tags()
+	}
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		var err error
+		resp, err = conn.CreateWebACL(params)
+		if err != nil {
+			if isAWSErr(err, wafv2.ErrCodeWAFUnavailableEntityException, "") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		_, err = conn.CreateWebACL(params)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error creating WAFv2 WebACL: %s", err)
```

Comment:
We prefer wrapping the original error, using the new [error wrapping in Go 1.13](https://blog.golang.org/go1.13-errors#TOC_3.3.), rather than just the string value of the original error. Please also update other error messages